tests/core/test_test.py

```diff
@@ -105,9 +143,24 @@ async def test_testing_valid_with_non_e2e_core_model(core_agent: Agent):
 
 
 async def _train_rule_based_agent(
-    moodbot_domain: Domain, train_file_name: Path
+    moodbot_domain: Domain,
+    train_file_name: Path,
+    monkeypatch: MonkeyPatch,
+    ignore_action_unlikely_intent: bool,
 ) -> Agent:
 
+    # We need `RulePolicy` to predict the correct actions
+    # in a particular conversation context as seen during training.
+    # Since it can get affected by `action_unlikely_intent` being triggered in
```

Comment:
I don't really understand this. Does this not also cause issues using the `RulePolicy` during normal prediction?

Reply:
@joejuzl 
I think I can explain this better.

Firstly, we are using `RulePolicy` as an example policy to get determinism in actions predicted from the ensemble. We don't want to worry about any other action being incorrectly predicted and only want to test that `action_unlikely_intent` is correctly handled by `rasa test`. 

Now, `RulePolicy` is the only policy which does not ignore `action_unlikely_intent`'s presence in the tracker. So, for a lot of tests that are written / modified in this PR, we cannot use `RulePolicy` as it is. We want it to ignore it in some circumstances otherwise we really cannot test all the test cases reliably.

During normal prediction, this situation would have never come up in the first place because:
1. An `action_unlikely_intent` is always only predicted right after the `UserUttered` event. It won't be predicted as a follow-up action ever.
2. If a rule can be used to make a prediction after `UserUttered` event then `action_unlikely_intent` will always be dropped (priorities inside ensemble).

So, in reality rules will always work as expected during normal prediction.

Does that clarify the need for all the monkey patching here? 😅

Reply:
Thanks for the explanation. 

> Now, RulePolicy is the only policy which does not ignore action_unlikely_intent's presence in the tracker.

Why is this the case?

Reply:
Other policies are trained only on stories and `action_unlikely_intent` cannot be included in a training story because it is un-learnable. Hence, they ignore `action_unlikely_intent` during inference as well.

However, you can include `action_unlikely_intent` as a [rule to customize what should happen when `action_unlikely_intent` is triggered](https://github.com/RasaHQ/rasa/pull/8913/files#diff-507719c0f3f45420d32a4427131fb37c24cff2bdbd9868410cab35ad6552557aR197). Since, `RulePolicy` is supposed to handle everything rules it cannot ignore `action_unlikely_intent`'s presence in the tracker.